ghproxy/ghmetrics/ghpath.go

```diff
@@ -187,19 +198,49 @@ func v(fragment string, children ...node) node {
 	}
 }
 
+func mv(fragment string, children ...node) node {
+	return node{
+		PathFragment: multiVariable(fragment),
+		children:     children,
+	}
+}
+
 func resolve(parent node, path []string) (string, bool) {
 	if !parent.Matches(path[0]) {
 		return "", false
 	}
 	representation := parent.Represent()
 	if len(path) == 1 || len(parent.children) == 0 {
+		representation = replaceMultiVariable(representation)
 		return representation, true
 	}
 	for _, child := range parent.children {
 		suffix, matched := resolve(child, path[1:])
 		if matched {
+			representation = replaceMultiVariable(representation)
 			return strings.Join([]string{representation, suffix}, "/"), true
 		}
 	}
+
+	if representation != "" && representation[0] == '+' {
```

Comment:
What if we don't embed the `PathFragment` in a `node` but instead allow a `node` to expose its own `Matches()`?

We could have a `func Matches(path []string) (matches bool, segments int)` that would allow a  `node` to match more than one path segment and return how many. Then a `node`'s `Matches()` could be defined as "match while the `path[0]` matches our `PathFragment` and no children match `path[0]`" or something? Just to make the actual algorithm in `resolve` simpler to reason about?

Reply:
I like the idea, would make it easier for sure. I'll hack on this tomorrow.